worlds/sc2wol/PoolFilter.py

```diff
@@ -82,9 +97,20 @@ def move_mission(mission_name, current_pool, new_pool):
             move_mission("Welcome to the Jungle", MissionPools.MEDIUM, MissionPools.EASY)
             move_mission("Engine of Destruction", MissionPools.HARD, MissionPools.MEDIUM)
 
+    remove_final_mission_from_other_pools(mission_pools)
     return mission_pools
 
 
+def remove_final_mission_from_other_pools(mission_pools: Dict[MissionPools, List[str]]):
+    final_missions = mission_pools[MissionPools.FINAL]
+    for pool, missions in mission_pools.items():
+        if pool == MissionPools.FINAL:
+            continue
+        for final_mission in final_missions:
+            while final_mission in missions:
+                missions.remove(final_mission)
```

Comment:
This is a lot of excess work, I suggest keeping a list of `goal_missions` as `SC2Mission` values and then doing something like this:
```
for mission in goal_missions:
    mission_pools[mission.pool].remove(mission.name)
    mission_pools[MissionPools.FINAL].append(mission.name)
```
Even better would be reusing `move_mission`, though that would require some extra refactoring.

On a separate note, since you use `while` to remove the mission: Is it possible and/or desired to have duplicates of missions in any pool? Maybe the pools should be sets instead of lists?

Reply:
The thing is if the mission moves between pools

Reply:
duplicates shouldn't exist